go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -400,6 +400,16 @@ func allocatePodAnnotationWithRollback(
 		hasIPAMClaim = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
 	}
 
+	defer func() {
+		if ipamClaim == nil || claimsReconciler == nil {
+			return
+		}
+		updatedClaim := claimsReconciler.UpdateIPAMClaimStatus(ipamClaim, podAnnotation, pod.Name, err)
+		if reconcileErr := claimsReconciler.Reconcile(ipamClaim, updatedClaim, ipAllocator); reconcileErr != nil {
```

Comment:
One more thing....
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/9ea5b14aa3b2dcf2e6ebf36bbf3330337cf262e1/go-controller/pkg/persistentips/allocator.go#L90-L104
Isn't the reconcile only updating the object when an IP was not already set?

Reply:
yes.
Once an IPAMClaim has IPs allocated, Reconcile will never update it via the API. It only validates that any subsequent reconciliation attempts are trying to use the same IPs.
This means:
- A malicious pod cannot wipe an existing IPAMClaim's IPs through your new code
- The only time IPs get written is when the claim is initially empty
- After that, Reconcile is read-only (validation-only)